tldextract/remote.py

```diff
@@ -3,16 +3,9 @@
 from __future__ import annotations
 
 import re
-from collections.abc import Callable
 from ipaddress import AddressValueError, IPv6Address
 from urllib.parse import scheme_chars
 
-inet_pton: Callable[[int, str], bytes] | None
-try:
-    from socket import AF_INET, AF_INET6, inet_pton  # Availability: Unix, Windows.
```

Comment:
I forget, was `inet_pton` for a faster implementation? This PR sacrifices those for correctness?

Reply:
Yes, it was for faster implementation, but its behavior is platform specific. We just learned that macOS still allows leading zeroes in ipv4 addresses today.